pkg/webhook/resources/storageclass/validator.go

```diff
@@ -106,3 +133,69 @@ func (v *storageClassValidator) validateSetUniqueDefault(newObj runtime.Object)
 
 	return nil
 }
+
+func (v *storageClassValidator) validateEncryption(newObj runtime.Object) error {
+	newSC := newObj.(*storagev1.StorageClass)
+
+	// Use util.LonghornOptionEncrypted as the key to check if the storage class is encrypted
+	if value, ok := newSC.Parameters[util.LonghornOptionEncrypted]; !ok {
+		return nil
+	} else if value != "true" {
+		return werror.NewInvalidError(fmt.Sprintf("storage class %s is not for encryption or decryption", newSC.Name), fmt.Sprintf("spec.parameters[%s] must be true", util.LonghornOptionEncrypted))
+	}
+
+	var (
+		secretName      string
+		secretNamespace string
+	)
+
+	for _, pair := range pairs {
+		name, nameExists := newSC.Parameters[pair[0]]
+		namespace, namespaceExists := newSC.Parameters[pair[1]]
+
+		if !nameExists || !namespaceExists {
+			missingParam := pair[0]
+			if !namespaceExists {
+				missingParam = pair[1]
+			}
+			return werror.NewInvalidError(fmt.Sprintf("storage class must contain %s", missingParam), "spec.parameters")
+		}
+
+		if secretName == "" && secretNamespace == "" {
+			secretName = name
+			secretNamespace = namespace
+		} else if secretName != name || secretNamespace != namespace {
+			return werror.NewInvalidError(fmt.Sprintf("secret names and namespaces in %s and %s are different from others", pair[0], pair[1]), "")
+		}
+	}
```

Comment:
how about this to get rid of unnecessary else if
```suggestion
		if secretName == "" && secretNamespace == "" {
			secretName = name
			secretNamespace = namespace
			continue
		}	
		if secretName != name || secretNamespace != namespace {
			return werror.NewInvalidError(fmt.Sprintf("secret names and namespaces in %s and %s are different from others", pair[0], pair[1]), "")
		}
	}
```